### PR TITLE
fix(Anticheat): Firetruck Flag Fix

### PR DESCRIPTION
**Pull request by Savage912-Official**
Issue: #1351

### CODE REVIEW

**Comment by minipunch** [comment]
Gonna mark as a draft since not tested

**Comment by Savage912-Official** [comment]
Tested and seems to be working as intended.

File: resources/usa_rp2/client/cl_weaponDamageModifier.lua [hunk 1]
```diff
@@ -78,21 +78,26 @@ end)
 -- 'modified weapon damage' anti cheat
 Citizen.CreateThread(function()
   local lastCheck = 0
+  local me = PlayerPedId()
+  local veh = GetVehiclePedIsIn(me, false)
```
Comment (minipunch) [inline resources/usa_rp2/client/cl_weaponDamageModifier.lua:82]: Calling natives (like `GetVehiclePedIsIn`) is performance intensive, and doing it every other frame (i.e., `Wait(1)` in a while true loop) is not good for optimization

Let's only call it when we need it 

So move it down to like under the check on like 85 so it gets called less and only when required

**Review by minipunch — CHANGES_REQUESTED** [review]

File: resources/usa_rp2/client/cl_weaponDamageModifier.lua (hunk 1 above)
Comment (Savage912-Official) [reply resources/usa_rp2/client/cl_weaponDamageModifier.lua:82]: Working on this now

Comment (Savage912-Official) [reply resources/usa_rp2/client/cl_weaponDamageModifier.lua]: Submitted change

**Review by minipunch — COMMENTED** [review]
Did u actually test to confirm this worked?

Cuz I am wondering if line 100 is actually the line being called when throwing the flag or if this will actually fix it since that alert is thrown in more than just the lines u changed

File: resources/usa_rp2/client/cl_weaponDamageModifier.lua (hunk 1 above)
Comment (minipunch) [reply resources/usa_rp2/client/cl_weaponDamageModifier.lua]: Looks better nice

**Review by minipunch — APPROVED** [review]

**Pull request merged**